src/prefect/environments/storage/docker.py

```diff
@@ -453,19 +462,30 @@ def create_dockerfile_object(self, directory: str) -> str:
         Returns:
             - str: the absolute file path to the Dockerfile
         """
+        # Either load the base commands from the specified dockerfile or define a FROM
+        if self.dockerfile:
+            with open(self.dockerfile, "r") as contents:
+                base_commands = contents.read()
+        else:
+            base_commands = "FROM {base_image}".format(base_image=self.base_image)
+
         # Generate single pip install command for python dependencies
         pip_installs = "RUN pip install "
         if self.python_dependencies:
             for dependency in self.python_dependencies:
                 pip_installs += "{} ".format(dependency)
 
+        # Write all extra commands that should be run in the image
+        installation_commands = ""
+        for cmd in self.installation_commands:
+            installation_commands += "RUN {}\n".format(cmd)
+
         # Generate ENV variables to load into the image
         env_vars = ""
         if self.env_vars:
-            white_space = " " * 20
-            env_vars = "ENV " + " \\ \n{}".format(white_space).join(
-                "{k}={v}".format(k=k, v=v) for k, v in self.env_vars.items()
-            )
+            # Format with repr to get proper quoting
+            formatted_vars = [f"{k}={v!r}" for k, v in self.env_vars.items()]
+            env_vars = "ENV " + " \\\n    ".join(formatted_vars)
```

Comment:
Functional change for environment variables, includes relative indent but not indent dependent on template indentation further down